integration-tests/kubernetes-client/src/test/java/io/quarkus/it/kubernetes/client/KubernetesClientSerializationTest.java

```diff
@@ -0,0 +1,144 @@
+    void serialization() {
+        given()
+                .contentType("application/json")
+                .body("""
+                        {
+                          "apiVersion": "v1",
+                          "kind": "Pod",
+                          "metadata": {
+                            "name": "serialization-test"
+                          },
+                          "spec": {
+                            "containers": [
+                              {
+                                "name": "test",
+                                "image": "busybox",
+                                "volumeMounts": [
+                                  {
+                                    "name": "test",
+                                    "mountPath": "/test"
+                                  }
+                                ]
+                              }
+                            ]
+                          }
+                        }
+                        """)
+                .when()
+                .put("/pod/{namespace}/{podName}", mockServer.getClient().getNamespace(), "serialization-test")
+                .then()
+                .statusCode(200);
+        // JSON data is persisted into the mock server
+        assertThat(mockServer.getClient().pods().withName("serialization-test").get())
+                .hasFieldOrPropertyWithValue("metadata.name", "serialization-test")
+                .extracting("spec.containers").asInstanceOf(InstanceOfAssertFactories.list(Container.class))
+                .singleElement()
+                .hasFieldOrPropertyWithValue("name", "test")
+                .hasFieldOrPropertyWithValue("image", "busybox")
+                .extracting(Container::getVolumeMounts)
+                .asInstanceOf(InstanceOfAssertFactories.list(VolumeMount.class))
+                .singleElement()
+                .hasFieldOrPropertyWithValue("name", "test")
+                .hasFieldOrPropertyWithValue("mountPath", "/test");
+    }
+
+    @Test
+    @DisplayName("GET - Should deserialize the Pod into valid JSON")
+    void deserialization() {
+        mockServer.getClient().pods().resource(new PodBuilder()
+                .withMetadata(new ObjectMetaBuilder()
+                        .withName("deserialization-test")
+                        .build())
+                .withSpec(new PodSpecBuilder()
+                        .addToContainers(new ContainerBuilder()
+                                .withName("deserialization-test-container")
+                                .addToPorts(new ContainerPortBuilder()
+                                        .withContainerPort(8080)
+                                        .build())
+                                .build())
+                        .build())
+                .build())
+                .create();
+        // JSON data is retrieved from the mock server
+        when()
```

Comment:
Is this really verifying that the JSON shape is the expected one and not the one with all the empty objects/arrays? https://github.com/quarkusio/quarkus/issues/39934#issuecomment-2119773533

Reply:
AFAIK, it should.
In any case, if you want to verify manually you can change line 113 from
```java
                .log().ifValidationFails(LogDetail.BODY);
```
to
```java
                .log().body();
```

We can also add another assertion to verify that the response body doesn't actually contain the word `overhead`:
```java
.body(not(containsString("overhead"))
```